src/Extensions/Diagnostics.HealthChecks/Internal/RestClientWrapper.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using Microsoft.ServiceFabric.Client;
+using Microsoft.ServiceFabric.Client.Http;
+
+namespace Microsoft.Omex.Extensions.Diagnostics.HealthChecks
+{
+	internal class RestClientWrapper
+	{
+		private ServiceFabricHttpClient m_client;
+
+		private readonly Uri m_clusterEndpoint;
+
+		private bool m_ready;
+
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+		public RestClientWrapper(Uri clusterEndpoint)
+#pragma warning restore CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+		{
+			// m_client is initialized during first publish async in order to avoid possible deadlocks
+			m_clusterEndpoint = clusterEndpoint;
+			m_ready = false;
+		}
+
+		public ServiceFabricHttpClient Get()
```

Comment:
This method should have `async` modifier (https://docs.microsoft.com/en-us/dotnet/csharp/async) then you would be able to use `await` keyword to get the result of the `Task` instead of calling `GetAwaiter` directly.